packages/myst-cli/src/project/utils.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Compute the indentation of a YAML line
+ *
+ * @param line
+ */
+export function yamlLineIndent(line: string): number | undefined {
+  const prefix = line.match(/^\s*[^\s#]/)?.[0];
+  return prefix ? prefix.length - 1 : undefined;
+}
+
+/**
+ * Test if line matches YAML section declaration
+ *
+ * @param name - section name
+ * @param line
+ *
+ **/
+function matchesYAMLSection(name: string, line: string): boolean {
+  // NB this is not escaped!
+  return !!line.match(new RegExp(`^\\s*${name}:`));
+}
+
+/**
+ * Find the start of a YAML section body with a given key.
+ *
+ * @param name - section name
+ * @param indent - indent of parent section
+ * @param lines - lines to parse
+ * @param index - index of starting line
+ *
+ * e.g. for the following YAML:
+ *   saturn:
+ *     best-moons:
+ *       - titan
+ *       - dione
+ *   jupiter:
+ *     best-moons:
+ *       - io
+ *       - europa
+ * This function can be used to find the lines bounding the `jupiter`- section, i.e.
+ *     best-moons:
+ *       - io
+ *       - europa
+ */
+export function findYAMLSection(
```

Comment:
Are there tests for this, this seems like a place where we want to have a lot of edge cases covered.